obspy/clients/httpproxy.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+"""
+Establish a socket connection through an HTTP proxy.
+Author: Fredrik Østrem <frx.apps@gmail.com>
+License:
+  This code can be used, modified and distributed freely, as long as it is this
+  note containing the original author, the source and this license, is put
+  along with the source code.
+
+J. MacCarthy, modified from https://gist.github.com/frxstrem/4487802
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+from future import standard_library
+from future.utils import native_str
+
+from base64 import b64encode
+import socket
+with standard_library.hooks():
+    import urllib
+    import urlparse
+
+
+def get_proxy_tuple():
+    """
+    Return system http proxy as a urlparse.urlparse tuple or () if unset.
+    """
+    proxydict = urllib.getproxies()
+    proxystr = proxydict.get('http') or proxydict.get('https') or ''
+    if proxystr:
+        proxy = urlparse.urlparse(proxystr)
+    else:
+        proxy = ()
+
+    return proxy
+
+
+def valid_address(addr):
+    """ Verify that an IP/port tuple is valid """
+    is_valid = (isinstance(addr, (list, tuple)) and
+                len(addr) == 2 and
+                isinstance(addr[0], (str, native_str)) and
+                isinstance(addr[1], int))
+    return is_valid
+
+
+def http_proxy_connect(address, proxy, auth=None, headers={}):
+    """
+    Establish a socket connection through an HTTP proxy.
+
+    Arguments:
+      address (required)     = The address of the target
+      proxy (def: None)      = The address of the proxy server
```

Comment:
I think you might have changed the wrong one; `proxy` is required, `auth` is optional.